Code/Framework/AzCore/AzCore/std/ranges/ranges_algorithm.h

```diff
@@ -1658,9 +1729,39 @@ namespace AZStd::ranges
                 bool_constant<indirectly_copyable<I1, O>>
                 >, copy_backward_result<I1, O>>
             {
-                for (I1 iter{ last }; iter != first;)
+                if constexpr (can_use_memcpy<I1, O>)
                 {
-                    *--result = *--iter;
+                    // Specialized copy for contiguous iterators which are trivially copyable
+                    if (const size_t numElements = ranges::distance(first, last); numElements > 0)
+                    {
+#if az_has_builtin_memmove
+                        result -= numElements;
+                        __builtin_memmove(to_address(result), to_address(first), numElements * sizeof(iter_value_t<I1>));
+#else
+                        if (az_builtin_is_constant_evaluated())
+                        {
+                            for (I1 iter{ last }; iter != first;)
+                            {
+                                *--result = *--iter;
```

Comment:
suggestion: This syntax is pretty confusing, would it possible to break up the statements slightly? The assembly should be the same it'll just be a lot easier to grok at a glance

Reply:
(repeated in a couple of spots)

Reply:
This code is similar to the possible implementatin that users would see on [cppreference](https://en.cppreference.com/w/cpp/algorithm/ranges/copy#Possible_implementation) for ranges::copy.

Reply:
Ah okay fair enough, it just looks super cryptic to me 😅 Leaving as-is is fine 👍